### PR TITLE
New branch PR

### PR DESCRIPTION
**Pull request by JustinObanor**

### CODE REVIEW

File: kufar/books/main.go [hunk 1]
```diff
@@ -88,39 +90,43 @@ func New() (*Database, error) {
...
 //CreateBook ...
 func (d Database) CreateBook(w http.ResponseWriter, r *http.Request) {
 	var bk Book
 
-	err := json.NewDecoder(r.Body).Decode(&bk)
-	if err != nil {
-		fmt.Printf("Error unmarshalling json: %v", err)
+	if err := json.NewDecoder(r.Body).Decode(&bk); err != nil {
+		http.Error(w, http.StatusText(http.StatusInternalServerError)+"Error unmarshalling json", http.StatusInternalServerError)
 	}
```
Comment (bynov) [inline kufar/books/main.go:113]: Please don't forget to call `return` here :)

File: kufar/books/main.go [hunk 2]
```diff
@@ -143,9 +149,8 @@ func (d Database) ReadBooks(w http.ResponseWriter, r *http.Request) {
...
 
 	w.Header().Set("Content-Type", "application/json")
-	err = json.NewEncoder(w).Encode(resps)
-	if err != nil {
-		fmt.Printf("Error marshalling json: %v", err)
+	if err := json.NewEncoder(w).Encode(resps); err != nil {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 	}
```
Comment (bynov) [inline kufar/books/main.go:156]: Same here

File: kufar/books/main.go [hunk 3]
```diff
@@ -154,11 +159,11 @@ func (d Database) ReadBook(w http.ResponseWriter, r *http.Request) {
 	id := mux.Vars(r)["id"]
 	idInt, err := strconv.Atoi(id)
 	if err != nil {
-		fmt.Printf("cant convert string: %v", err)
+		log.Printf("cant convert string: %v", err)
```
Comment (bynov) [inline kufar/books/main.go]: Please add `http.Error` and `return`

File: kufar/books/main.go [hunk 4]
```diff
@@ -171,15 +176,14 @@ func (d Database) ReadBook(w http.ResponseWriter, r *http.Request) {
...
 	}
 
 	resp := convertToResponse(bk)
 	w.Header().Set("Content-Type", "application/json")
-	err = json.NewEncoder(w).Encode(resp)
-	if err != nil {
-		fmt.Printf("Error marshalling json: %v", err)
+	if err := json.NewEncoder(w).Encode(resp); err != nil {
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 	}
```
Comment (bynov) [inline kufar/books/main.go:191]: `return` here

File: kufar/books/main.go [hunk 5]
```diff
@@ -188,46 +192,45 @@ func (d Database) UpdateBook(w http.ResponseWriter, r *http.Request) {
 	id := mux.Vars(r)["id"]
 	idInt, err := strconv.Atoi(id)
 	if err != nil {
-		fmt.Printf("cant convert string: %v", err)
+		log.Printf("cant convert string: %v", err)
 	}
```
Comment (bynov) [inline kufar/books/main.go:201]: `http.Error` and `return`

File: kufar/books/main.go [hunk 6]
```diff
@@ -188,46 +192,45 @@ func (d Database) UpdateBook(w http.ResponseWriter, r *http.Request) {
...
+	w.Write([]byte(http.StatusText(http.StatusOK)))
 }
 
 //DeleteBook ...
 func (d Database) DeleteBook(w http.ResponseWriter, r *http.Request) {
 	id := mux.Vars(r)["id"]
 	idInt, err := strconv.Atoi(id)
 	if err != nil {
-		fmt.Printf("cant convert string: %v", err)
+		log.Printf("cant convert string: %v", err)
```
Comment (bynov) [inline kufar/books/main.go]: `http.Error` and `return`

**Review by bynov — CHANGES_REQUESTED** [review]

File: kufar/books/main.go (hunk 6 above)
Comment (JustinObanor) [reply kufar/books/main.go]: Sure, ill change it. Thought that since we're just converting a string. we didn't need to use `http.Error`

Comment (bynov) [reply kufar/books/main.go]: @JustinObanor  Nope, it's a BadRequest error because your API expect `int` here

Comment (JustinObanor) [reply kufar/books/main.go]: @bynov Alright then

**Review by bynov — APPROVED** [review]
Noice!

giphy

**Pull request merged**